test/plugins/test-trace-hapi.js

```diff
@@ -308,6 +308,41 @@ describe('hapi', function() {
           });
         });
       });
+
+      it('should end spans when client aborts request', function(done) {
```

Comment:
I think if you're calling `res.end` on your own, you shouldn't change whether you call it or not based on client behavior.

hapi (I'm looking [here](https://github.com/hapijs/hapi/blob/09c2e3374cf0ad63144f90268fb0b0d0e8a2ceba/lib/transmit.js#L249)) seems to not call it if the request was aborted, suggesting that it might be a no-op.